File: back/src/main/java/com/back/domain/concert/dto/ConcertDetailResponse.java
package com.back.domain.concert.dto;

import com.back.domain.concert.entity.Concert;

import java.util.List;
import java.util.Map;

public record ConcertDetailResponse(
        Long concertId,
        String concertName,
        String description,
        String venueName,
        String location,
        List<String> detailUrlList,
        Map<String, Integer> prices,
        boolean bookable
) {
    public static ConcertDetailResponse of(
            Concert concert,
            String venueName,
            String location,
            List<String> detailUrlList,
            Map<String, Integer> prices,
            boolean bookable
    ) {
        return new ConcertDetailResponse(
                concert.getConcertId(),
                concert.getConcertName(),
                concert.getDescription(),
                venueName,
                location,
                detailUrlList,
                prices,
                bookable
        );
    }
}

File: back/src/main/kotlin/com/back/domain/concert/dto/ConcertDetailResponse.kt
package com.back.domain.concert.dto

import com.back.domain.concert.entity.Concert

data class ConcertDetailResponse(
    val concertId: Long,
    val concertName: String,
    val description: String?,
    val venueName: String,
    val location: String,
    val urlPoster: String?,
    val detailUrlList: List<String>,
    val prices: Map<String, Int>,
    val bookable: Boolean,
    val reviewSummary: String?
) {
    companion object {
        fun of(
            concert: Concert,
            venueName: String,
            location: String,
            detailUrlList: List<String>,
            prices: Map<String, Int>,
            bookable: Boolean
        ): ConcertDetailResponse = ConcertDetailResponse(
            concertId = checkNotNull(concert.concertId) { "Concert ID null" },
            concertName = concert.concertName,
            description = concert.description,
            venueName = venueName,
            location = location,
            urlPoster = concert.urlPoster,
            detailUrlList = detailUrlList,
            prices = prices,
            bookable = bookable,
            reviewSummary = concert.reviewSummary
        )
    }
}
